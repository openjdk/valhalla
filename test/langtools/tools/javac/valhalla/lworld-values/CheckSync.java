/*
 * @test /nodynamiccopyright/
 * @summary May not synchronize on value types
 *
 * @compile/fail/ref=CheckSync.out -XDrawDiagnostics CheckSync.java
 */

/* Note: ATM, value types do not have jlO in their lineage. So they anyway
   cannot synchronize using the methods declared on jlO.
*/

public final primitive class CheckSync {

    final primitive class Val {
        int x = 10;
        void foo() {
            // All calls below are bad.
            wait();
            wait(10);
            wait(10, 10);
            notify();
            notifyAll();
            finalize();
            clone();
        }
    }

    final Val val = Val.default;

    void test() throws InterruptedException {
        // All calls below are bad.
        val.wait();
        val.wait(10);
        val.wait(new Integer(10));
        val.wait(new Long(10));
        val.wait(10L);
        val.wait(10L, 10);
        val.notify();
        val.notifyAll();
    }
    int x = 10;
}
