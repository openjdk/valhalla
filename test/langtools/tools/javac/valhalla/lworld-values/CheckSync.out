CheckSync.java:18:17: compiler.err.primitive.class.does.not.support: wait
CheckSync.java:19:17: compiler.err.primitive.class.does.not.support: wait
CheckSync.java:20:17: compiler.err.primitive.class.does.not.support: wait
CheckSync.java:21:19: compiler.err.primitive.class.does.not.support: notify
CheckSync.java:22:22: compiler.err.primitive.class.does.not.support: notifyAll
CheckSync.java:23:21: compiler.err.primitive.class.does.not.support: finalize
CheckSync.java:24:18: compiler.err.primitive.class.does.not.support: clone
CheckSync.java:32:17: compiler.err.primitive.class.does.not.support: wait
CheckSync.java:33:17: compiler.err.primitive.class.does.not.support: wait
CheckSync.java:34:18: compiler.warn.has.been.deprecated.for.removal: java.lang.Integer(int), java.lang.Integer
CheckSync.java:34:17: compiler.err.primitive.class.does.not.support: wait
CheckSync.java:35:18: compiler.warn.has.been.deprecated.for.removal: java.lang.Long(long), java.lang.Long
CheckSync.java:35:17: compiler.err.primitive.class.does.not.support: wait
CheckSync.java:36:17: compiler.err.primitive.class.does.not.support: wait
CheckSync.java:37:17: compiler.err.primitive.class.does.not.support: wait
CheckSync.java:38:19: compiler.err.primitive.class.does.not.support: notify
CheckSync.java:39:22: compiler.err.primitive.class.does.not.support: notifyAll
- compiler.note.deprecated.filename: CheckSync.java
- compiler.note.deprecated.recompile
15 errors
2 warnings
